src/styles/text/text.js

```diff
@@ -305,18 +335,33 @@ Object.assign(TextStyle, {
         // Process labels by priority
         let priorities = Object.keys(labels).sort((a, b) => a - b);
         for (let priority of priorities) {
-            if (!labels[priority]) {
+            if (!labels[priority]) { // no labels at this priority, skip to next
                 continue;
             }
 
+            // i is a label object at the current priority level, example:
+            // Object: {
+            //  area: undefined,
+            //  feature: Object,
+            //  label: LabelLine,
+            //  style: "100 24px Helvetica/rgb(102,102,102)/rgb(255,255,255)/8"
+            // }
+
             for (let i = 0; i < labels[priority].length; i++) {
                 let { style, feature, label } = labels[priority][i];
 
+                // test the label for intersections with other labels in the tile
                 if (!label.discard(this.aabbs[tile])) {
+                    // if it didn't collide (or won a collision)
```

Comment:
There is not really a concept of winning a collision, not colliding would be sufficient.